secboot/secboot_sb.go

```diff
@@ -456,3 +456,40 @@ func (key *SealKeyRequest) getWriter() (sb.KeyDataWriter, error) {
 		return key.BootstrappedContainer.GetTokenWriter(key.SlotName)
 	}
 }
+
+// ConvertOldDisk takes on disk using legacy keyslots 0, 1, 2 and add
+// names to those keyslots. This is needed to convert the save disk
+// during a factory reset.
+func ConvertOldDisk(devicePath string) error {
+	if err := sb.NameLegacyLUKS2ContainerKey(devicePath, 0, "old-default-key"); err != nil && !errors.Is(err, sb.KeyslotAlreadyHasANameErr) {
```

Comment:
this does nothing if the slot has already a name right?

Reply:
Yes. A "legacy container key" does not have a name.